Monika-After-Story/MonikaModDev
# One more love song for Monika to sing

Down to one song from three, but also includes changes to the song db framework.

## Changes

- Adds the song 'Stand By Me'

- the long version of a song will now automatically unlock when the short counterpart is first played

- the song selector now separates the songs into 2 menus, long and short

- the functions `getUnlockedSongs()` and `hasUnlockedSongs()` can now both take in a `length` parameter for short and long songs

## Testing

- run the update script for v0_10_2

- with no long songs unlocked (you may have to manually lock `monika_aiwfc`) goto `Hey, [m_name]>music>Can you sing me a song?` and verify you are not asked what kind of song you want to hear.

- wait for `monika_song_stand_by_me` to play randomly (can call `monika_sing_song_random` to speed this up) and verify the short version plays.

- goto `Hey, [m_name]>music>Can you sing me a song?` and verify Monika asks you what kind of song you'd like to hear and that when you pick a Full song, 'Stand By Me' is unlocked and plays the full version.

- verify that when in the song selector with both long and short songs available that you can switch between lists at will
